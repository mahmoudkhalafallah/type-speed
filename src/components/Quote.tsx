import React from 'react'
import styled from "styled-components"

const StyledQuote = styled.div`
width: 100%;
padding: 15px;
font-size: 20px;
background: #e7e7e7;
user-select: none;
`

const GreenText = styled.span`
color: green;
`

const RedText = styled.span`
background: #e34d4d;
color: #fff;
`

const ActiveText = styled.span`
text-decoration: underline;
`

interface Props {
    data: string[];
    index: number;
    length: number;
    isValid: boolean;
}

const Quote: React.FC<Props> = ({ data, index, length, isValid }) => <StyledQuote>
    {data.length && data.slice(0, index).join(" ")}
    {(index > 0) && " "}
    {data[index] && (isValid ? <GreenText>{data[index].slice(0, length)}</GreenText> : <RedText>{data[index].slice(0, length)}</RedText>)}
    {data[index] && <ActiveText>{data[index].slice(length)}</ActiveText>}
    {" "}
    {data.length && data.slice(index + 1).join(" ")}
</StyledQuote>

export default Quote