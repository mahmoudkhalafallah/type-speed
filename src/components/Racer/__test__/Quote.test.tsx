import React from 'react'
import Quote from '../Quote'
import { render } from '@testing-library/react'

const str = 'Pork rump minim, laborum spare ribs pancetta lorem ut meatloaf ullamco swine sint voluptate pariatur. Ullamco mollit consequat est. Esse brisket spare ribs ipsum eiusmod t-bone mollit veniam. Exercitation short loin filet mignon, tenderloin tail anim kevin ribeye minim beef velit esse salami. Rump et do landjaeger.'.split(' ')

test('Quote rendered correctly', () => {
    const { baseElement } = render(
        <Quote data={str} index={0} length={0} isValid={false} />,
    )

    expect(baseElement).toMatchSnapshot()
})

test('Quote is valid', () => {
    const { baseElement } = render(
        <Quote data={str} index={0} length={1} isValid={true} />,
    )

    expect(baseElement).toMatchSnapshot()
})

test('Quote is invalid', () => {
    const { baseElement } = render(
        <Quote data={str} index={0} length={1} isValid={false} />,
    )

    expect(baseElement).toMatchSnapshot()
})