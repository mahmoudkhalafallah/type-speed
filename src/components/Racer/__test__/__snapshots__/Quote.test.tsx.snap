// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote is invalid 1`] = `
<body>
  <div>
    <div
      class="sc-bdVaJa gnKtJD"
    >
      
      <span
        class="sc-htpNat joxNFM"
      >
        P
      </span>
      <span
        class="sc-bxivhb GkQco"
      >
        ork
      </span>
       
      rump minim, laborum spare ribs pancetta lorem ut meatloaf ullamco swine sint voluptate pariatur. Ullamco mollit consequat est. Esse brisket spare ribs ipsum eiusmod t-bone mollit veniam. Exercitation short loin filet mignon, tenderloin tail anim kevin ribeye minim beef velit esse salami. Rump et do landjaeger.
    </div>
  </div>
</body>
`;

exports[`Quote is valid 1`] = `
<body>
  <div>
    <div
      class="sc-bdVaJa gnKtJD"
    >
      
      <span
        class="sc-bwzfXH kRBeZh"
      >
        P
      </span>
      <span
        class="sc-bxivhb GkQco"
      >
        ork
      </span>
       
      rump minim, laborum spare ribs pancetta lorem ut meatloaf ullamco swine sint voluptate pariatur. Ullamco mollit consequat est. Esse brisket spare ribs ipsum eiusmod t-bone mollit veniam. Exercitation short loin filet mignon, tenderloin tail anim kevin ribeye minim beef velit esse salami. Rump et do landjaeger.
    </div>
  </div>
</body>
`;

exports[`Quote rendered correctly 1`] = `
<body>
  <div>
    <div
      class="sc-bdVaJa gnKtJD"
    >
      
      <span
        class="sc-htpNat joxNFM"
      />
      <span
        class="sc-bxivhb GkQco"
      >
        Pork
      </span>
       
      rump minim, laborum spare ribs pancetta lorem ut meatloaf ullamco swine sint voluptate pariatur. Ullamco mollit consequat est. Esse brisket spare ribs ipsum eiusmod t-bone mollit veniam. Exercitation short loin filet mignon, tenderloin tail anim kevin ribeye minim beef velit esse salami. Rump et do landjaeger.
    </div>
  </div>
</body>
`;
